arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -204,6 +204,17 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
 
   AgencyCache::databases_t plan = clusterInfo.getPlan(planIndex, dirty);
 
+  if (plan.empty()) {
+    // this early exit is a safeguard against overzealous cleanup by 
+    // the maintenance. in case the plan is empty and compared to the
+    // local state, the maintenance may decide to drop everyting 💥
+    LOG_TOPIC("0a6f2", DEBUG, Logger::MAINTENANCE)
+        << "DBServerAgencySync::execute no plan change set";
+    result.success = true;
+    result.errorMessage = "DBServerAgencySync::execute no reconciliation plan";
```

Comment:
Shouldn't that be
```suggestion
    result.success = false;
    result.errorMessage = "DBServerAgencySync::execute no reconciliation plan";
```
instead?

Reply:
Or is it considered successful in the sense, that it has done everything it should on the plan that's available?